### ISSUE TITLE
Floating point comparison test failure

### ISSUE THREAD

**Issue by shterrett** [georust/rust-geo#75]
On current master:

```
---- algorithm::distance::test::line_segment_distance_test stdout ----
        thread 'algorithm::distance::test::line_segment_distance_test' panicked at 'assertion failed: `(left == right)` (left: `2.048590078926335`, right: `2.0485900789263356`)', src/algorithm/distance.rs:194
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

The difference is that the last digit of `left` is not included in `right`. Exact comparison of floating points is always somewhat iffy - what do you think about moving to a "close-enough" comparison?

I can implement the sample code from here: http://floating-point-gui.de/errors/comparison/ as a function, and replace the `assert_eq!` with `assert!(float_eq(actual, expected))`

**Comment by mbattifarano** [comment]
fwiw this happens to me as well.

**Comment by urschrei** [comment]
I agree, an almost_equals function is a great idea here.

**Comment by frewsxcv** [comment]
Would the implementation be anything more than:

```rust
fn almost_equals<F: Float>(f1, f2) -> bool {
   f1.abs_sub(f2) < f1::epsilon()
}
```

**Comment by TeXitoi** [comment]
In this particular case, I suspect that the test will still fail. `epsilon` must be scaled. Something like that:
```rust
fn almost_equals<F: Float>(f1: F, f2: F) -> bool {
    f1.abs_sub(f2) < F::epsilon() * F::max(f1, f2) * F::from(1e3).unwrap()
}
```
for forgetting the 3 last digits.

**Comment by frewsxcv** [comment]
A PR was opened for this: https://github.com/georust/rust-geo/pull/82

Anyone have thoughts?

**Comment by frewsxcv** [comment]
Closed with #82
